Ageless Item Fixes

Patrick12115 opened the pull request:
Fixes issues with Mirror shield not showing and MS sword and sheath showing when Link does not have a sword equipped

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh.otr.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2051661644.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2051697555.zip)
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2051701926.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/2051702783.zip)
<!--- section:artifacts:end -->

Patrick12115 committed:
Fixes Mirror Shield and MS sheath

inspectredc commented:
Could you explain a bit more what this is trying to fix? I'm not quite sure of the issue

Patrick12115 commented:
> Could you explain a bit more what this is trying to fix? I'm not quite sure of the issue

It is a fix for the Ageless items that was already merged. Some edge cases I missed and something I evidently changed late in the PR and never fully tested. Mirror shield wasn't showed properly when it should and there was a case with the master sword sheath that I missed.

inspectredc commented:
Would it be possible to show some before + after images? cause im still not sure what test cases to check 😅

Patrick12115 commented:
> Would it be possible to show some before + after images? cause im still not sure what test cases to check 😅 
 

I'm currently at work, but if you load up current develop, turn the options on, there was a case where mirror shield wasn't showing as child link. I can't remember exactly which case at this current moment, but it was obvious, whichever it was. And the other one is adult link shielding with the deku shield and kokiri sword was showing master sword sheath, I do believe. But if you're still not sure by time I get home I will show comparisons

Patrick12115 committed:
Added Hookshot Tip&Chain and more edge cases

Patrick12115 committed:
Update presets.h

Patrick12115 committed:
Revert "Update presets.h"

This reverts commit 95d952ba23cf4dba1b0eb9299df7e1f9e12a4763.

Patrick12115 committed:
More fixes

Bow showing up for slingshot gallery minigame is now fixed, as well as a rouge instance shrinking links hand when I didn't mean for it to

Patrick12115 committed:
Merge remote-tracking branch 'upstream/develop' into Ageless-Items-Fixes

Patrick12115 committed:
Merge remote-tracking branch 'upstream/develop' into Ageless-Items-Fixes

Patrick12115 committed:
CVar stuffs

Patrick12115 committed:
Fixes

Pepper0ni commented:
Master sword still appears while shielding if link has a non-sword, non-empty thing on B (example: sticks).

Patrick12115 committed:
Merge branch 'develop' into Ageless-Items-Fixes

Patrick12115 committed:
Fix more sheath stuff and bgs hand

Extloga commented on soh/soh/SohMenuBar.cpp:
```diff
@@ -1192,6 +1192,15 @@ void DrawEnhancementsMenu() {
                 "- Blue Warps\n"
                 "- Darunia\n"
                 "- Gold Skulltulas");
+            if (UIWidgets::PaddedEnhancementCheckbox("Show Age-Dependent Equipment", CVAR_ENHANCEMENT("EquimentAlwaysVisible"), true,
```
Typo: `Equiment` &#8594; `Equipment`

This one appears multiple times in the added `CVAR_ENHANCEMENT` brackets.

Malkierian commented:
Fix the typos Extloga brought up and I'll get this merged.

Patrick12115 committed:
Merge branch 'develop' into Ageless-Items-Fixes

Patrick12115 commented:
> Fix the typos Extloga brought up and I'll get this merged.

Done

Patrick12115 committed:
Add the P
